pinecone/index_connection.go

```diff
@@ -387,9 +387,16 @@ func (idx *IndexConnection) ListVectors(ctx context.Context, in *ListVectorsRequ
 		vectorIds[i] = &res.Vectors[i].Id
 	}
 
+	var usage *Usage
+	if res.Usage != nil {
+		usage = &Usage{ReadUnits: derefOrDefault(res.Usage.ReadUnits, 0)}
+	} else {
+		usage = nil
+	}
+
 	return &ListVectorsResponse{
 		VectorIds:           vectorIds,
-		Usage:               &Usage{ReadUnits: derefOrDefault(res.Usage.ReadUnits, 0)},
+		Usage:               usage,
```

Comment:
This was actually a nil pointer reference bug discovered from testing against Pinecone Local. Based on the spec and struct, it's possible for `Usage` to be nil, and we weren't properly taking that into account here.

Reply:
good catch